src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes/HttpContextRefactor/HttpContextRefactorCodeFixProvider.cs

```diff
@@ -0,0 +1,297 @@
+            var semantic = await context.Document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (semantic is null)
+            {
+                return;
+            }
+
+            // Find the type declaration identified by the diagnostic.
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+            if (semantic.GetOperation(node, context.CancellationToken) is not IPropertyReferenceOperation property)
+            {
+                return;
+            }
+
+            var methodOperation = GetEnclosingMethodOperation(property);
+
+            if (methodOperation is null)
+            {
+                return;
+            }
+
+            //// Register a code action that will invoke the fix.
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: CodeFixResources.HttpContextRefactorTitle,
+                    createChangedSolution: c => InjectHttpContext(context.Document, methodOperation, property, c),
+                    equivalenceKey: nameof(CodeFixResources.HttpContextRefactorTitle)),
+                diagnostic);
+        }
+
+        private async Task<Solution> InjectHttpContext(Document document, IOperation methodOperation, IPropertyReferenceOperation propertyOperation, CancellationToken cancellationToken)
+        {
+            var slnEditor = new SolutionEditor(document.Project.Solution);
+
+            var editor = await slnEditor.GetDocumentEditorAsync(document.Id, cancellationToken).ConfigureAwait(false);
+
+            // Add parameter if not available
+            var parameter = await GetOrAddMethodParameterAsync(editor, document, methodOperation, propertyOperation, cancellationToken).ConfigureAwait(false);
+
+            if (parameter is null)
+            {
+                return document.Project.Solution;
+            }
+
+            // Update node usage
+            editor.ReplaceNode(propertyOperation.Syntax, parameter);
+
+            if (methodOperation.SemanticModel?.GetDeclaredSymbol(methodOperation.Syntax, cancellationToken) is ISymbol methodSymbol)
+            {
+                await UpdateCallersAsync(methodOperation.SemanticModel, methodSymbol, propertyOperation.Property, slnEditor, cancellationToken).ConfigureAwait(false);
+            }
+
+            return slnEditor.GetChangedSolution();
+        }
+
+        private IOperation? GetEnclosingMethodOperation(IOperation? operation)
+            => operation.GetParentOperation(IsEnclosedMethodOperation);
+
+        private async Task<SyntaxNode?> GetOrAddMethodParameterAsync(DocumentEditor editor, Document document, IOperation methodOperation, IPropertyReferenceOperation propertyOperation, CancellationToken token)
+        {
+            // Get the symbol representing the type to be renamed.
+            var semanticModel = await document.GetSemanticModelAsync(token).ConfigureAwait(false);
```

Comment:
is this the same semantic model used above?

Reply:
Good catch. Passing it through...